services/brig/src/Brig/API/OAuth.hs

```diff
@@ -314,31 +300,24 @@ lookupOauthClient cid = do
     q :: PrepQuery R (Identity OAuthClientId) (OAuthApplicationName, RedirectUrl)
     q = "SELECT name, redirect_uri FROM oauth_client WHERE id = ?"
 
-lookupOAuthClientSecret :: (MonadClient m) => OAuthClientId -> m (Maybe Password)
-lookupOAuthClientSecret cid = do
-  runIdentity <$$> retry x5 (query1 q (params LocalQuorum (Identity cid)))
-  where
-    q :: PrepQuery R (Identity OAuthClientId) (Identity Password)
-    q = "SELECT secret FROM oauth_client WHERE id = ?"
-
-insertOAuthAuthorizationCode :: (MonadClient m) => Word64 -> OAuthAuthorizationCode -> OAuthClientId -> UserId -> OAuthScopes -> RedirectUrl -> m ()
-insertOAuthAuthorizationCode ttl code cid uid scope uri = do
+insertOAuthAuthorizationCode :: (MonadClient m) => Word64 -> OAuthAuthorizationCode -> OAuthClientId -> UserId -> OAuthScopes -> RedirectUrl -> OAuthCodeChallenge -> m ()
+insertOAuthAuthorizationCode ttl code cid uid scope uri chal = do
   let cqlScope = C.Set (Set.toList (unOAuthScopes scope))
-  retry x5 . write q $ params LocalQuorum (code, cid, uid, cqlScope, uri)
+  retry x5 . write q $ params LocalQuorum (code, cid, uid, cqlScope, uri, chal)
   where
-    q :: PrepQuery W (OAuthAuthorizationCode, OAuthClientId, UserId, C.Set OAuthScope, RedirectUrl) ()
-    q = fromString $ "INSERT INTO oauth_auth_code (code, client, user, scope, redirect_uri) VALUES (?, ?, ?, ?, ?) USING TTL " <> show ttl
+    q :: PrepQuery W (OAuthAuthorizationCode, OAuthClientId, UserId, C.Set OAuthScope, RedirectUrl, OAuthCodeChallenge) ()
+    q = fromString $ "INSERT INTO oauth_auth_code (code, client, user, scope, redirect_uri, code_challenge) VALUES (?, ?, ?, ?, ?, ?) USING TTL " <> show ttl
```

Comment:
I know it's from before, but why is `ttl` not part of the query parameters? It is everywhere else in the code.